### ISSUE TITLE
Refactor how column families and it's fields are maintained in ORM table class

### ISSUE THREAD

**Issue by thhermansen** [CompanyBook/massive_record#13]
Now, I really feel the urge to refactor how ColumnFamilies and it's
fields are maintained in the ORM. It's interface is hard to use unless
you use it via the default column_family :name do; <fields>; end
construction in a class.

It feels bad for the following reasons:

```
1.  ORM-objects should have column_families like today, but it should
    return an object (a set) of column families. Not an array
    of ColumnFamilies which we have to loop over
    and detect to find one column. This new ColumnFamilies
    object should provide a simple interface to get-or-create-new column
    family given by name etc.
2.  Adding a field to an existing column family a hassle is today. The
    following is required and needs to be done manually every place you
    want to add a field to a family:
      1: Find (or create) your column family (detect on the
          column families array).
      2: Update ORM-object's attributes_schema hash to reflect new fields.
      3: Call undefine_attribute_methods to make write/read
          methods be regenerated.
    The responsibility for doing what needs to be done when a field is
    added to a column family should be done by the column family self.
3.  As mentioned in number two; attributes_schema, which now is a
    separate hash linking attributes to fields should be fetched from
    the ColumnFamilies object. As it is now, we need to maintain things
    more than one place, which is bad.
```

I know that it was written to be used with the following interface:
    class Test < Table
      column_family :info do
        field :an_attribute
      end
    end

..but now we have two places which injects fields to families dynamically (IdFactory and Devise::Orm::MassiveRecord (glue for making MassiveRecord work with Devise)). Maybe I have the time to look at this soon :-) ..anyone having any other thoughts on this?

**Comment by thhermansen** [comment]
Progress has been made on this, see comparison between develop and my refactor branch: https://github.com/CompanyBook/massive_record/compare/develop...feature/refactor_schema

I think these changes are good, and in short we now have:
- Made ColumnFamilies into a set; it is no longer possible to have column families with the same name.
- Made Fields into a set; it is no longer possible to have fields with the same name.
- The enforcement of fields with unique names are global across all column families.
- The attributes_schema() is read from from the column family set, and there are no longer a need to maintain multiple hashes which reflected the column families and fields in different ways.
- Added simple interface for dynamically adding fields to a family: ATableClass.add_field_to_column_family(:info, :name, :default => "NA"). It will take care of keeping the ATableClass consistent with read/write and other attribute methods it creates based on the schema.
- ORM::Column is now implemented, mening that it should be easier to subclass it, and define fields given the DSL we have defined in the README.
- ORM::Column has attributes_schema(), known_attribute_names(), default_attributes_from_schmea() just like the ORM::Table. In fact, these methods comes from the same module.
- All old tests passes, and I suspect that we should not need to do a lot of changing in the Front. What we _can_ do in the front is clean up classes which inherits from ORM::Column, as we can now define fields for these classes as well :-)
